### PR TITLE
traceroute/traceroute6: Cleanup usage and manual fixes

### PR DESCRIPTION
**Pull request by jlduran**
The following are a series of minor fixes, intended to reduce the differences between the two utilities, in the event of a possible future unification.

### CODE REVIEW

**Comment by jlduran** [comment]
> [!NOTE]
> Given the differences between our code in `contrib` and the actual upstream sources have diverged considerably.

**Comment by jlduran** [comment]
Note to self: Croos-builds are failing for aarch64-clang-12 after 9eecef052155646fbc5f8f533b952b372572d06a, due to clang-12 being too old?

File: contrib/traceroute/ifaddrlist.c [hunk 1]
```diff
@@ -141,7 +141,7 @@ ifaddrlist(register struct ifaddrlist **ipaddrp, register char *errbuf)
 			continue;
 
 
-		(void)strncpy(device, ifr.ifr_name, sizeof(ifr.ifr_name));
+		(void)strncpy(device, ifr.ifr_name, sizeof(device) - 1);
```
Comment (bsdimp) [inline contrib/traceroute/ifaddrlist.c:144]: if you are changing this, you'd be better off using strlcpy and nerfing the next line.

File: contrib/traceroute/traceroute.8 [hunk 2]
```diff
@@ -129,9 +126,22 @@ and supported by the peer)
...
 at the destination host (so an ICMP PORT_UNREACHABLE message will
-be returned to terminate the route tracing).  If something is
+be returned to terminate the route tracing).
+If something is
 listening on a port in the default range, this option can be used
 to pick an unused port range.
+.It Fl P Ar proto
+Send packets of specified IP protocol.
+The currently supported protocols
+are: UDP, UDP-Lite, TCP, SCTP, GRE and ICMP.
```
Comment (bsdimp) [inline contrib/traceroute/traceroute.8:136]: you could join these two lines I think...

File: contrib/traceroute/traceroute.8 [hunk 3]
```diff
@@ -148,27 +158,35 @@ that has no route through it (e.g., after the interface was dropped by
 .Xr routed 8 .
 .It Fl s Ar src_addr
 Use the following IP address (which usually is given as an IP number, not
-a hostname) as the source address in outgoing probe packets.  On
+a hostname) as the source address in outgoing probe packets.
+On
 multi-homed hosts (those with more than one IP
```
Comment (bsdimp) [inline contrib/traceroute/traceroute.8:163]: You can join these two lines.

File: contrib/traceroute/traceroute.8 [hunk 4]
```diff
@@ -148,27 +158,35 @@ that has no route through it (e.g., after the interface was dropped by
...
 .It Fl t Ar tos
 Set the
 .Em type-of-service
-in probe packets to the following value (default zero).  The value must be
-a decimal integer in the range 0 to 255.  This option can be used to
-see if different types-of-service result in different paths.  The upper six
-bits are the Differentiated Services Codepoint (RFC4594). The lower two
+in probe packets to the following value (default zero).
+The value must be
+a decimal integer in the range 0 to 255.
```
Comment (bsdimp) [inline contrib/traceroute/traceroute.8:180]: join

File: contrib/traceroute/traceroute.8 [hunk 5]
```diff
@@ -148,27 +158,35 @@ that has no route through it (e.g., after the interface was dropped by
...
+This option can be used to
+see if different types-of-service result in different paths.
+The upper six
```
Comment (bsdimp) [inline contrib/traceroute/traceroute.8:183]: maybe you can join this and the next line.

**Comment by bsdimp** [comment]
So this generally looks good, but the manual page could use a little attention as noted wrt line breaks. If you are really ambitious, you could re-wrap the lines now broken (the join comments I made is a minimal subset of this).

Otherwise, I like it. You now have your commit bit and if you do the above, you can mark this as 'Reviewed-by: imp' when you commit it :)

**Comment by jlduran** [comment]
Will do!

**Comment by jlduran** [comment]
The man pages were originally wrapped at 72 CPL. Old school! Felt a little sad extending it to 80 chars.
Thank you for the review!

**Comment by bsdimp** [comment]
looks good now, minor tweaks to one or two of the commit messagesg.

**Pull request closed without merging**